spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -605,7 +605,8 @@ public static void importSparkPartitions(
           filesToImport
               .map((MapFunction<DataFile, String>) f -> f.path().toString(), Encoders.STRING())
               .toDF("file_path");
-      Dataset<Row> existingFiles = loadMetadataTable(spark, targetTable, MetadataTableType.ENTRIES);
+      Dataset<Row> entriesFiles = loadMetadataTable(spark, targetTable, MetadataTableType.ENTRIES);
+      Dataset<Row> existingFiles = entriesFiles.filter(entriesFiles.col("status").notEqual(2));
```

Comment:
I wanted to use `ManifestEntry.Status.DELETED` instead but looks like the class is not public per https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/ManifestEntry.java#L31